### PR TITLE
Bug 1845561: enable resource requests for config reloaders

### PR DESCRIPTION
**Pull request by simonpasquier**
Follow-up of #938. Needs https://github.com/openshift/prometheus-operator/pull/94 to be merged first.

* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: jsonnet/prometheus-user-workload.jsonnet [hunk 1]
```diff
@@ -256,21 +256,15 @@ local alertmanagerRole =
...
-            //      memory: '5Mi',
-            //      cpu: '1m',
-            //    },
-            //  },
-            //},
+            {
+              name: 'config-reloader',
+              resources: {
+                requests: {
+                  memory: '5Mi',
```
Comment (simonpasquier) [inline jsonnet/prometheus-user-workload.jsonnet:263]: @s-urbaniak not sure if we still want to set memory requests here. If yes, which value should we pick?
`5Mi` is the previous value when we used `jimmidyson/configmap-reload`. Looking at the numbers from the `e2e-aws-operator` job, `prometheus-config-reloader` is a bit more demanding.

image

Comment (s-urbaniak) [reply jsonnet/prometheus-user-workload.jsonnet:263]: looking at the graph `10Mi` seems like a reasonable value to me.

Comment (simonpasquier) [reply jsonnet/prometheus-user-workload.jsonnet:263]: :+1:

second question: should we apply the same memory request to all config requests (Alertmanager, prometheus-k8s and Thanos ruler)? I'll also gather data from a running cluster to corroborate.

Comment (s-urbaniak) [reply jsonnet/prometheus-user-workload.jsonnet:263]: if they expose the same memory consumption, I would say so.

Comment (simonpasquier) [reply jsonnet/prometheus-user-workload.jsonnet:263]: `10Mi` seems about right.

image

**Comment by simonpasquier** [comment]
PR updated setting memory request to `10Mi` for all config-reloader containers.

File: jsonnet/prometheus-user-workload.jsonnet (hunk 1 above)
Comment (s-urbaniak) [reply jsonnet/prometheus-user-workload.jsonnet]: perfect

**Comment by s-urbaniak** [comment]
@simonpasquier does it make sense to bump the resource requests in kube-prometheus too?

**Pull request merged**